src/actions/Follow.ts

```diff
@@ -0,0 +1,70 @@
+import { commands, DocumentLink, env, Uri, window } from "vscode";
+import {
+  Action,
+  ActionPreferences,
+  ActionReturnValue,
+  Graph,
+  TypedSelection,
+} from "../typings/Types";
+import displayPendingEditDecorations from "../util/editDisplayUtils";
+import { ensureSingleTarget } from "../util/targetUtils";
+
+export default class Follow implements Action {
+  getTargetPreferences: () => ActionPreferences[] = () => [
+    { insideOutsideType: "inside" },
+  ];
+
+  constructor(private graph: Graph) {
+    this.run = this.run.bind(this);
+  }
+
+  async run([targets]: [TypedSelection[]]): Promise<ActionReturnValue> {
+    const target = ensureSingleTarget(targets);
+
+    await displayPendingEditDecorations(
+      targets,
+      this.graph.editStyles.referenced
+    );
+
+    const link = await this.findLink(target);
+    if (link) {
+      await this.openUri(link.target!);
+    } else {
+      await this.graph.actions.executeCommand.run(
+        [targets],
+        "editor.action.revealDefinition",
+        { restoreSelection: false }
+      );
+    }
+
+    return {
+      thatMark: targets.map((target) => target.selection),
+    };
+  }
+
+  private async findLink(target: TypedSelection) {
+    const links = <DocumentLink[]>(
+      await commands.executeCommand(
+        "vscode.executeLinkProvider",
+        target.selection.editor.document.uri
+      )
+    );
+    return links.find((link) =>
+      link.range.contains(target.selection.selection)
+    );
+  }
+
+  private async openUri(uri: Uri) {
+    switch (uri.scheme) {
+      case "http":
+      case "https":
+        await env.openExternal(uri);
+        break;
+      case "file":
+        await window.showTextDocument(uri);
```

Comment:
If we're not able to get a link to a specific line, then that's a problem.  Or do these URIs somehow link to a specific line? If not, maybe we should fall back to the makeshift `editor.action.revealDefinition` implementation if we detect a file URI?

Reply:
A uri does not include a line number. But definitions are not returned as a line
`file://a.md` is the kind of things that will be returned as a file link

Reply:
Ah ok so definitions don't get returned from `executeLinkProvider`?  If not, then I think this approach is reasonable

Reply:
Correct